rampwf/score_types/detection.py

```diff
@@ -751,6 +819,28 @@ def average_precision_interpolated(ps, rs):
     return ap
 
 
+def average_precision_exact(ps, rs):
+    # from https://github.com/amdegroot/ssd.pytorch/blob/ce4c994db0ee11f82aabb4fdb3499dc970156db5/eval.py#L182-L213
+
+    # correct AP calculation
+    # first append sentinel values at the end
+    mrec = np.concatenate(([0.], rs, [1.]))
+    mpre = np.concatenate(([0.], ps, [0.]))
+
```

Comment:
Sure this is not a `[1.]` after `ps, ` ?

Reply:
Hmm, good question. I copied it from https://github.com/amdegroot/ssd.pytorch/blob/ce4c994db0ee11f82aabb4fdb3499dc970156db5/eval.py#L182-L213, which of course does not mean it is correct.

Intuitively indeed it should be different for precision and recall

Reply:
I suppose that because a few lines further, you take the cumulative max along the precision curve, it does not matter (or to say, the 0 is replace by the highest obtained precision, which is not necessarily 1 if the first prediction (the one with the highest confidence) is not a good one)

Reply:
same seems to be done by google in tensorflow: https://github.com/tensorflow/models/blob/fb12693f2420cc687d0ca8c640c241610b2b036b/research/object_detection/utils/metrics.py#L110-L120

Reply:
> I suppose that because a few lines further, you take the cumulative max along the precision curve, it does not matter (or to say, the 0 is replace by the highest obtained precision, which is not necessarily 1 if the first prediction (the one with the highest confidence) is not a good one)

👍  sounds like a good reason why